Automattic/wp-calypso
# Refactor the bundle installation to be generic

Related to https://github.com/Automattic/themes/issues/7498

First part of this refactor: https://github.com/Automattic/wp-calypso/pull/84483

## Proposed Changes

* Refactor the bundle installation to be generic.
  * Everything that happens for WooCommerce should continue having exactly the same behavior.
  * It now should support other bundles.
* Remove Woo Verify Email, which was not used, always protected by a feature flag.
  * PR 1 with the feature: https://github.com/Automattic/wp-calypso/pull/64387
  * PR 2 with the feature: https://github.com/Automattic/wp-calypso/pull/64462
  * Commit: https://github.com/Automattic/wp-calypso/pull/84886/commits/ad225e6fc52d699e6d6725acda1875bd20e3472d
  * Slack conversation: p1701903126481889-slack-C0Q664T29
* Remove unused code from `site-setup-flow`.
  * Commit: https://github.com/Automattic/wp-calypso/pull/84886/commits/d8ea69e997811d0e3c2d4bb1a419796ba967a2fc
  * Slack conversation: p1702302906398119-slack-C0Q664T29

## Testing Instructions



The following tests are just to make sure everything continues working as previously. Nothing should change and anything new was introduced. Also, please test other different scenarios that you can think of.

**Flow starting with free plan and selecting a WooCommerce theme**

1. Navigate to `http://calypso.localhost:3000/start`.
2. Select the free plan.
3. On the "Pick a design" step, go to the "store" tab, and select the "Tsubaki" theme (WooCommerce theme).
4. Continue through the flow, upgrading the plan to be able to install the theme.
5. Make sure everything works in the same way as previously (a good way I used to compare was to navigate in wordpress.com in parallel), and when you complete all the steps you are redirected to the WooCommerce home with WC installed.

**Flow starting with business plan and selecting a WooCommerce theme**

1. Navigate to `http://calypso.localhost:3000/start`.
2. Select the business plan.
3. On the "Pick a design" step, go to the "store" tab, and select the "Tsubaki" theme (WooCommerce theme).
4. Continue through the flow.
5. Make sure everything works in the same way as previously (a good way I used to compare was to navigate in wordpress.com in parallel), and when you complete all the steps you are redirected to the WooCommerce home with WC installed.

**Flow starting with free plan and free theme changing to WooCommerce theme later**

1. Navigate to `http://calypso.localhost:3000/start`.
2. Select the free plan.
3. On the "Pick a design" step, select the Twenty Twenty-Four theme.
4. Continue through the flow until redirected to the site.
5. On the site, navigate to Appearance > Themes.
7. Select the "Tsubaki" theme.
8. Upgrade, and continue through the flow.
9. Make sure everything works in the same way as previously (a good way I used to compare was to navigate in wordpress.com in parallel), and when you complete all the steps you are redirected to the WooCommerce home with WC installed.

**Flow starting with business plan and free theme, and changing to WooCommerce theme later**

1. Navigate to `http://calypso.localhost:3000/start`.
2. Select the business plan.
3. On the "Pick a design" step, select the Twenty Twenty-Four theme.
4. Continue through the flow.
5. Make sure everything works in the same way as previously (a good way I used to compare was to navigate in wordpress.com in parallel). Notice that in this case, WooCommerce wasn't installed. Not sure if it's the ideal behavior, but it keeps the current behavior. If confirmed that it's not an intentional behavior, I can open an issue for that.

## Known issues

* When I started developing this PR, I noticed an intermittent error (also reproduced on `trunk` and staging). When I completed the flow with a WooCommerce bundle (`woo-on-plans`), the _"Customize this design"_ button in `http://calypso.localhost:3000/marketplace/thank-you/{DOMAIN}` was always a link to the site editor. But sometimes when clicked, it was redirecting to WooCommerce home (`/wp-admin/admin.php?page=wc-admin`), and others it was just navigating to the site editor without the redirect. I didn't find where this redirect logic lives (I suspect it's not in Calypso), and I couldn't reproduce it anymore, so I'm ignoring it. If you are testing it, and see the same behavior, I can open an issue for that.

## Pre-merge Checklist
